sagemaker/24_train_bloom_peft_lora/sagemaker-notebook.ipynb

```diff
@@ -50,34 +86,39 @@
    "source": [
     "import sagemaker\n",
     "import boto3\n",
-    "sess = sagemaker.Session()\n",
-    "# sagemaker session bucket -> used for uploading data, models and logs\n",
-    "# sagemaker will automatically create this bucket if it not exists\n",
-    "sagemaker_session_bucket=None\n",
-    "if sagemaker_session_bucket is None and sess is not None:\n",
-    "    # set to default bucket if a bucket name is not given\n",
-    "    sagemaker_session_bucket = sess.default_bucket()\n",
+    "from botocore.exceptions import NoCredentialsError, BotoCoreError\n",
     "\n",
     "try:\n",
+    "    sess = sagemaker.Session()\n",
     "    role = sagemaker.get_execution_role()\n",
-    "except ValueError:\n",
+    "    sagemaker_session_bucket = sess.default_bucket()\n",
+    "except (NoCredentialsError, BotoCoreError) as e:\n",
+    "    print(f\"Exception occurred while trying to fetch default execution role and bucket: {e}\")\n",
+    "    # If an exception occurs while fetching the default execution role, try getting the `sagemaker_execution_role`\n",
     "    iam = boto3.client('iam')\n",
-    "    role = iam.get_role(RoleName='sagemaker_execution_role')['Role']['Arn']\n",
-    "\n",
-    "sess = sagemaker.Session(default_bucket=sagemaker_session_bucket)\n",
+    "    try:\n",
+    "        role = iam.get_role(RoleName='sagemaker_execution_role')['Role']['Arn']\n",
+    "    except (NoCredentialsError, BotoCoreError) as e:\n",
+    "        print(f\"Exception occurred while trying to fetch `sagemaker_execution_role`: {e}\")\n",
+    "        # Handle this exception or re-raise it, based on your needs\n",
+    "        role = None  # set role as None if unable to fetch it\n",
     "\n",
     "print(f\"sagemaker role arn: {role}\")\n",
-    "print(f\"sagemaker bucket: {sess.default_bucket()}\")\n",
-    "print(f\"sagemaker session region: {sess.boto_region_name}\")\n"
+    "if role is not None:\n",
+    "    print(f\"sagemaker bucket: {sess.default_bucket()}\")\n",
+    "    print(f\"sagemaker session region: {sess.boto_region_name}\")\n"
```

Comment:
thats our common snippet we use, since we test the notebooks outside of studio

Reply:
we don't need to be over specific here.

Reply:
and whats the point of catching and error and then printing and not letting the cell fail.